server/http/s2s_realtime.cpp

```
@@ -0,0 +1,947 @@
+        throw std::invalid_argument("PCM16 audio must contain complete samples");
+    std::vector<float> samples(bytes.size() / 2);
+    for (size_t i = 0; i < samples.size(); ++i) {
+        const uint16_t bits = static_cast<uint8_t>(bytes[i * 2]) |
+                              (static_cast<uint16_t>(static_cast<uint8_t>(bytes[i * 2 + 1])) << 8);
+        samples[i] = static_cast<int16_t>(bits) / 32768.0f;
+    }
+    return samples;
+}
+
+int16_t
+float_to_pcm16(float sample) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)